From c41b4a6742abc62d0d6108163324621d0c9d5ba7 Mon Sep 17 00:00:00 2001
Message-Id: <c41b4a6742abc62d0d6108163324621d0c9d5ba7.1423172493.git.feitong.yi@intel.com>
In-Reply-To: <18bb7324f916bffbc75e07ea0ffd3e6e68e7b6d8.1423172493.git.feitong.yi@intel.com>
References: <18bb7324f916bffbc75e07ea0ffd3e6e68e7b6d8.1423172493.git.feitong.yi@intel.com>
From: Deepak M <m.deepak@intel.com>
Date: Tue, 13 Jan 2015 12:49:32 +0530
Subject: [PATCH 07/29] MUST_REBASE [VPG]: drm/i915: Add wrapper function to
 get platform gen id

This wrapper function gets the gen id from which we can decide
whether it is a baytrail or cherrytrail device.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed.

Issue: GMINL-5193
Change-Id: I87a8ba2eba06fa25ad5c16da1d45f08280cf5dc4
Signed-off-by: Deepak M <m.deepak@intel.com>
---
 drivers/gpu/drm/i915/i915_adf_wrapper.c |    9 +++++++++
 include/drm/i915_adf.h                  |    1 +
 2 files changed, 10 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_adf_wrapper.c b/drivers/gpu/drm/i915/i915_adf_wrapper.c
index b5d10c3..114a42e 100644
--- a/drivers/gpu/drm/i915/i915_adf_wrapper.c
+++ b/drivers/gpu/drm/i915/i915_adf_wrapper.c
@@ -173,4 +173,13 @@ u32 intel_adf_get_pwm_vbt_data(void)
 }
 EXPORT_SYMBOL(intel_adf_get_pwm_vbt_data);
 
+u8 intel_adf_get_platform_id(void)
+{
+	if (!i915_adf_dev)
+		return 0;
+
+	return i915_adf_dev->info.gen;
+}
+EXPORT_SYMBOL(intel_adf_get_platform_id);
+
 #endif
diff --git a/include/drm/i915_adf.h b/include/drm/i915_adf.h
index da9fb64..4349888 100644
--- a/include/drm/i915_adf.h
+++ b/include/drm/i915_adf.h
@@ -56,6 +56,7 @@ extern int intel_adf_context_on_event(void);
 extern void intel_adf_display_rpm_get(void);
 extern void intel_adf_display_rpm_put(void);
 extern int i915_adf_driver_initialize(void);
+extern u8 intel_adf_get_platform_id(void);
 #endif
 
 
-- 
1.7.9.5

